opendevin/controller/agent_controller.py

```diff
@@ -176,14 +187,24 @@ def get_agent_state(self):
     async def start_delegate(self, action: AgentDelegateAction):
         AgentCls: Type[Agent] = Agent.get_cls(action.agent)
         agent = AgentCls(llm=self.agent.llm)
+        state = State(
+            inputs=action.inputs or {},
+            iteration=self.state.iteration,
+            max_iterations=self.state.max_iterations,
+            num_of_chars=self.state.num_of_chars,
+            # TODO: the delegate probably doesn't need to know the full history
+            history=self.state.history,
```

Comment:
I believe we explicitly did _not_ share history here before, so that the delegate has a fresh start.

Reply:
I'd suggest removing this, unless you think it's necessary for now

Reply:
it's not necessary for now but it's useful sometimes... this delegation thing is a bit tricky indeed

Reply:
I recall why I did this. By design, some micro agents need to know the full picture. If we don't pass history they would cry and throw errors.

We could fix those agents though... whatever they need to know can be passed by inputs.